core/domain/story_domain.py

```diff
@@ -234,6 +234,37 @@ def require_valid_node_id(cls, node_id):
             raise utils.ValidationError(
                 'Invalid node_id: %s' % node_id)
 
+    @classmethod
+    def require_valid_thumbnail_filename(cls, thumbnail_filename):
+        """Checks whether the thumbnail filename of the node is a valid
+            one.
+
+        Args:
+            thumbnail_filename: str. The thumbnail filename to validate.
+        """
+        if thumbnail_filename is not None:
+            if not isinstance(thumbnail_filename, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Expected thumbnail filename to be a string, received %s'
+                    % thumbnail_filename)
+            if thumbnail_filename.rfind('.') == 0:
+                raise utils.ValidationError(
+                    'Thumbnail filename should not start with a dot.')
+            if '/' in thumbnail_filename or '..' in thumbnail_filename:
+                raise utils.ValidationError(
+                    'Thumbnail filename should not include slashes or '
+                    'consecutive dot characters.')
+            if '.' not in thumbnail_filename:
+                raise utils.ValidationError(
+                    'Thumbnail filename with no extension.')
```

Comment:
Follow the format of your other error messages: Thumbnail filename should include an extension.

Reply:
Done.